riscv_analysis/src/lints/instruction_in_text.rs

```diff
@@ -0,0 +1,91 @@
+use crate::{
+    cfg::{Cfg, Segment},
+    passes::{LintError, LintPass},
+};
+
+/// A lint to ensure that instructions only exist in the text
+/// segment.
+///
+/// Instructions will only be assembled if they appear in
+/// the text segment. Instructions in other locations is
+/// behaviour that we do not handle.
+pub struct InstructionInTextCheck;
+impl LintPass for InstructionInTextCheck {
+    fn run(cfg: &Cfg, errors: &mut Vec<LintError>) {
+        for node in &cfg.clone() {
+            if node.node().is_instruction() && node.segment != Segment::Text {
+                errors.push(LintError::InvalidSegment(node.node().clone()));
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{arith, directive, iarith, parser::ParserNode};
+
+    /// Helper test function to run a single pass along parser nodes for testing.
+    fn run_single_pass_along_nodes<T>(nodes: &[ParserNode]) -> Vec<LintError>
+    where
+        T: LintPass,
+    {
+        let cfg = Cfg::new(nodes.into()).unwrap();
+        let mut errors = Vec::new();
+        T::run(&cfg, &mut errors);
+        errors
+    }
```

Comment:
Is there some place we could put this function so that we can use it for testing other lints in the future?

Reply:
Moved to a trait method, the syntax is cleaner as well.